src/target/sam4l.c

```diff
@@ -368,22 +366,21 @@ static int sam4l_flash_write_buf(target_flash_s *f, target_addr_t addr, const vo
 		src_data++;
 	}
 	/* write the page */
-	if (sam4l_flash_command(t, page, FLASH_CMD_WP)) {
-		return -1;
-	}
-	return 0;
+	if (sam4l_flash_command(t, page, FLASH_CMD_WP))
```

Comment:
As this if is immediately followed by returns on both branches, we'd suggest it should be replaced by a direct `return !sam4l_flash_command();` (clang-tidy should say something to this effect too)